kakao/src/main/kotlin/com/agoda/kakao/screen/Screen.kt

```diff
@@ -51,11 +143,24 @@ open class Screen<out T: Screen<T>>: ScreenActions {
             })
         }
 
-        inline fun <reified T : Screen<T>> onScreen(function: T.() -> Unit): T {
+        /**
+         * Initializes instance of the screen class provided and invokes given tail lambda on it.
+         *
+         * This approach helps to reduce boilerplate code and avoid have a screen instance stored in
+         * a properties of you tests classes.
+         *
+         * In order to use this function, your [Screen] class must have an empty primary constructor.
+         *
+         * @param function Tail lambda to be invoked on the created instance of screen.
+         */
+        inline fun <reified T : Screen<T>> onScreen(noinline function: T.() -> Unit): T {
             return T::class.java
                     .newInstance()
-                    .also { it.rootView?.isVisible() }
-                    .apply { function.invoke(this) }
+                    .apply { this(function) }
         }
+
+        internal val vis: Stack<Interceptor<ViewInteraction, ViewAssertion, ViewAction>?> = Stack()
```

Comment:
Is it an appropriate place to set screen stacks?
I suggest to replace it in another class.

Reply:
We will still have to put it in some object, and since the `Screen` instances are manipulating it, it is only normal to place it to the `Screen`'s companion object.